hr_check_ssnid/models/hr_employee.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from stdnum import get_cc_module
+from stdnum.exceptions import (
+    InvalidChecksum,
+    InvalidComponent,
+    InvalidFormat,
+    InvalidLength,
+)
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    ssnid_key = fields.Char(
+        string='SSN No Key',
+        help='Social Security Number Key',
+        size=2,
+        required=True,
+    )
+
+    @api.onchange('ssnid', 'ssnid_key') # if these fields are changed, call method
```

Comment:
Shouldn't this be an `@api.constraint` ?
You can probably apply it directly in `check_ssnid` method.
Although remember `self` is going to be a recordset, so `for rec in self:`